### ISSUE TITLE
Gradient does not exist warning when using weave model

### ISSUE THREAD

**Issue by arunppsg** [deepchem/deepchem#3427]
```py
import deepchem as dc
tasks, datasets, transformers = dc.molnet.load_delaney(featurizer=dc.feat.WeaveFeaturizer())
train, valid, test = datasets
model = dc.models.WeaveModel(n_tasks=1, mode='regression')
model.fit(train, nb_epoch=1000)
```

When I execute the above code, I get the following warning:
```
WARNING:tensorflow:Gradients do not exist for variables ['weave_layer_1/kernel:0', 'weave_layer_1/Variable:0', 
'weave_layer_1/kernel:0', 'weave_layer_1/Variable:0', 'weave_layer_1/kernel:0', 'weave_layer_1/Variable:0', 
'weave_layer_1/batch_normalization_3/gamma:0', 'weave_layer_1/batch_normalization_3/beta:0', 
'weave_layer_1/batch_normalization_4/gamma:0', 'weave_layer_1/batch_normalization_4/beta:0', 
'weave_layer_1/batch_normalization_5/gamma:0', 'weave_layer_1/batch_normalization_5/beta:0'] when 
minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` 
```

I think this is some serious warning as without gradients, the model will not be able to learn properly.
